### PR TITLE
fix: process onboarding exams correctly in onboarding status API used by onboarding panel

### PR DESCRIPTION
**Pull request by MichaelRoytman**
**Description:**

When returning information about a learner's onboarding status in a course, we exclude onboarding exams that the Learning Sequences API marks as inaccessible. While processing the list of onboarding exams in the course, the list was being modified during iteration, leading to incorrect behavior. This code change fixes this behavior. This fix also exposed a bug in the corresponding tests, which was also fixed in this code change.

**JIRA:**

None.

**Pre-Merge Checklist:**

- [x] Updated the version number in `edx_proctoring/__init__.py` and `package.json` if these changes are to be released.
- [x] Described your changes in `CHANGELOG.rst`
- [x] Confirmed Github reports all automated tests/checks are passing.
- [ ] Approved by at least one additional reviewer.

**Post-Merge:**

- [ ] Create a tag matching the new version number.

### CODE REVIEW

File: edx_proctoring/views.py [hunk 1]
```diff
@@ -446,7 +447,10 @@ def get(self, request):
             )
 
             if usage_key not in user_outline.accessible_sequences:
-                onboarding_exams.remove(onboarding_exam)
+                inaccessible_onboarding_exams.append(onboarding_exam)
```
Comment (schenedx) [inline edx_proctoring/views.py:450]: Can you explain why this change fixes the bug?

Comment (MichaelRoytman) [reply edx_proctoring/views.py:450]: The original code is modifying a list that it is concurrently iterating over, resulting in an incorrect calculation for `onboarding_exams`.

For example, if you have a list `onboarding_exams` with a length of two, and you hit line 449 and remove an exam, the iteration stops short of the second onboarding exam.

Comment (schenedx) [reply edx_proctoring/views.py:450]: Got it. Good catch.

**Review by schenedx — APPROVED** [review]
👍
